parsons/empower/empower.py

```diff
@@ -0,0 +1,261 @@
+        Get Empower profiles.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["profiles"])
+        for col in ["createdMts", "lastUsedEmpowerMts", "updatedMts"]:
+            tbl.convert_column(col, lambda x: self._unix_convert(x))
+        tbl.remove_column("activeCtaIds")  # Get as a method via get_profiles_active_ctas
+        return tbl
+
+    def get_profiles_active_ctas(self):
+        """
+        Get active ctas assigned to Empower profiles.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["profiles"]).long_table("eid", "activeCtaIds")
+        return tbl
+
+    def get_regions(self):
+        """
+        Get Empower regions.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["regions"])
+        tbl.convert_column("inviteCodeCreatedMts", lambda x: self._unix_convert(x))
+        return tbl
+
+    def get_cta_results(self):
+        """
+        Get Empower call to action results.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["ctaResults"])
+        tbl.convert_column("contactedMts", lambda x: self._unix_convert(x))
+        tbl = tbl.unpack_nested_columns_as_rows(
+            "answerIdsByPromptId", key="profileEid", expand_original=True
+        )
+        tbl.unpack_list("answerIdsByPromptId_value", replace=True)
+        col_list = [v for v in tbl.columns if v.find("value") != -1]
+        tbl.coalesce_columns("answer_id", col_list, remove_source_columns=True)
+        tbl.remove_column("uid")
+        tbl.remove_column("answers")  # Per docs, this is deprecated.
+        return tbl
+
+    def _split_ctas(self):
+        """
+        Internal method to split CTA objects into tables.
+        """
+
+        ctas = Table(self.data["ctas"])
+        for col in [
+            "createdMts",
+            "scheduledLaunchTimeMts",
+            "updatedMts",
+            "activeUntilMts",
+        ]:
+            ctas.convert_column(col, lambda x: self._unix_convert(x))
+        ctas.remove_column("regionIds")  # Get as a table via get_cta_regions()
+        ctas.remove_column("shareables")  # Get as a table via get_cta_shareables()
+        ctas.remove_column("prioritizations")  # Get as a table via get_cta_prioritizations()
+        ctas.remove_column("questions")  # This column has been deprecated.
+        cta_prompts = ctas.long_table("id", "prompts", prepend=False, retain_original=False)
+        cta_prompts.remove_column("ctaId")
+        cta_prompt_answers = cta_prompts.long_table("id", "answers", prepend=False)
+
+        return [ctas, cta_prompts, cta_prompt_answers]
+
+    def get_ctas(self):
```

Comment:
Enough data restructuring is happening in these methods (or the private/utility methods they're calling) that it would be useful for each of the dcostrings for the public methods to describe the format of the data being returned.